src/argilla/client/sdk/datasets/v1/api.py

```diff
@@ -0,0 +1,301 @@
+#  coding=utf-8
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import lru_cache
+from typing import Any, Dict, List, Optional, Union
+
+import httpx
+
+from argilla.client.sdk.client import AuthenticatedClient
+from argilla.client.sdk.commons.errors_handler import handle_response_error
+from argilla.client.sdk.commons.models import (
+    ErrorMessage,
+    HTTPValidationError,
+    Response,
+)
+from argilla.client.sdk.datasets.v1.models import (
+    FeedbackDatasetModel,
+    FeedbackRecordsModel,
+)
+
+
+def create_dataset(
+    client: AuthenticatedClient,
+    name: str,
+    workspace_id: str,
+    guidelines: Optional[str] = None,
+) -> Response[Union[FeedbackDatasetModel, ErrorMessage, HTTPValidationError]]:
+    url = "{}/api/v1/datasets".format(client.base_url)
+
+    body = {"name": name, "workspace_id": workspace_id}
+    if guidelines is not None:
+        body.update({"guidelines": guidelines})
+
+    response = httpx.post(
+        url=url,
+        json=body,
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    if response.status_code == 201:
+        parsed_response = FeedbackDatasetModel(**response.json())
+        return Response(
+            status_code=response.status_code,
+            content=response.content,
+            headers=response.headers,
+            parsed=parsed_response,
+        )
+    return handle_response_error(response)
+
+
+@lru_cache(maxsize=None)
```

Comment:
we can remove this for now, and introduce a cache mechanism later. This method can be called several times and the invalidating the cache can be a hell

Reply:
Sure, I used it because it was already used in the former `datasets/api.py`